File: samples/hello.rb
# TruePink — Ruby
require "set"

MAX = 100

class Cat
  attr_reader :name

  def initialize(name = "Pinky")
    @name = name
  end

  def meow(loud: false)
    msg = "Hello, #{@name}"
    loud ? msg.upcase : msg
  end
end

cat = Cat.new("Mochi")
(0...MAX).each do |i|
  puts cat.meow(loud: true) if i.even?
end

File: samples/hello.php
<?php
// TruePink — PHP
namespace App;

const MAX = 100;

class Cat
{
    private string $name;

    public function __construct(string $name = "Pinky")
    {
        $this->name = $name;
    }

    public function meow(bool $loud = false): string
    {
        $msg = "Hello, {$this->name}";
        return $loud ? strtoupper($msg) : $msg;
    }
}

$cat = new Cat("Mochi");
for ($i = 0; $i < MAX; $i++) {
    if ($i % 2 === 0) {
        echo $cat->meow(true) . PHP_EOL;
    }
}
